build-tools/src/main/java/org/elasticsearch/gradle/plugin/GenerateNamedComponentsTask.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.plugin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.elasticsearch.gradle.plugin.scanner.ClassReaders;
+import org.elasticsearch.gradle.plugin.scanner.NamedComponentScanner;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.file.ProjectLayout;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.tasks.CompileClasspath;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.workers.WorkAction;
+import org.gradle.workers.WorkParameters;
+import org.gradle.workers.WorkerExecutor;
+import org.objectweb.asm.ClassReader;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+public abstract class GenerateNamedComponentsTask extends DefaultTask {
+    private static final Logger LOGGER = Logging.getLogger(GenerateNamedComponentsTask.class);
+    private static final String NAMED_COMPONENTS_FILE = "named_components.json";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```

Comment:
That OBJECT_MAPPER should live in the nested worker class IMO as its only used there